include/deal.II/base/tensor.h

```diff
@@ -2138,7 +2138,7 @@ DEAL_II_CONSTEXPR inline DEAL_II_ALWAYS_INLINE
   scalar_product(const Tensor<rank, dim, Number> &     left,
                  const Tensor<rank, dim, OtherNumber> &right)
 {
-  typename ProductType<Number, OtherNumber>::type result;
+  typename ProductType<Number, OtherNumber>::type result{};
```

Comment:
I am surprised that you had to modify this line as well. Did you get similar errors?

Reply:
Yes, this is the only other line I got an error on.